okio-files/src/commonTest/kotlin/okio/files/PathTest.kt

```diff
@@ -198,4 +199,68 @@ class PathTest {
     assertEquals("a/b", "a//b/./.".toPath().toString())
     assertEquals("a/b/c", "a/b/./c/".toPath().toString())
   }
+
+  @Test
+  fun `windows fully qualified path`() {
+    assertEquals("C:\\", "C:\\".toPath().toString())
+    assertEquals("C:\\Windows\\notepad.exe", "C:\\Windows\\notepad.exe".toPath().toString())
+  }
+
+  @Test
+  fun `composing windows path`() {
+    assertEquals("C:\\Windows\\notepad.exe".toPath(), "C:\\".toPath() / "Windows" / "notepad.exe")
+  }
+
+  @Test
+  fun `windows path traversal up`() {
+    assertEquals("C:\\z".toPath(), "C:\\x\\y\\..\\..\\..\\z".toPath())
+    assertEquals("C:..\\z".toPath(), "C:x\\y\\..\\..\\..\\z".toPath())
+  }
+
+  @Test
+  fun `volume letter`() {
+    assertEquals('C', "C:\\Windows".toPath("\\").volumeLetter)
+    assertEquals('C', "C:".toPath("\\").volumeLetter)
+    assertEquals('z', "z:\\Windows".toPath("\\").volumeLetter)
+    assertEquals('z', "z:".toPath("\\").volumeLetter)
+    assertNull("\\\\server\\directory".toPath("\\").volumeLetter)
+    assertNull("\\\\server\\directory".toPath("\\").volumeLetter)
+    assertNull("\\Windows".toPath("\\").volumeLetter)
+    assertNull("/Windows".toPath("\\").volumeLetter)
+    assertNull("\\".toPath("\\").volumeLetter)
+    assertNull("/".toPath("\\").volumeLetter)
+    assertNull("C:".toPath("/").volumeLetter)
+    assertNull("C:/Windows".toPath("/").volumeLetter)
+  }
+
+  @Test
+  fun `windows absolute path`() {
+    assertEquals("\\Program Files", "\\Program Files".toPath().toString())
+    assertEquals("\\Program Files\\Photoshop".toPath(), "\\Program Files".toPath() / "Photoshop")
+  }
+
+  /** If a volume is present in the relative path... */
+  @Test
+  fun `windows volume relative path`() {
+    assertEquals("c:notepad.exe".toPath("\\"), "A:\\DOOM".toPath() / "c:notepad.exe")
```

Comment:
Did DOOM use to fit on a floppy drive?

Reply:
That’s how I played it! https://www.ebay.com/b/Doom-PC-1993-Video-Games/139973/bn_43712974

Reply:
Amazing! First PC I had came with a CD-ROM